### ISSUE TITLE
Anaconda Error:  Autoformatting failed, buffer not changed.

### ISSUE THREAD

**Issue by neuro-01** [DamnWidget/anaconda#841]
Getting error in Sublime text 3, when I test autoformatting after saving a file:
> Autoformatting failed, buffer not changed.

Anaconda settings:
```
{
    "auto_formatting": true,
    "auto_formatting_timeout": 5,
    "autoformat_ignore":
    [
        "E309",
        "E501"
    ],
    "pep8_ignore":
    [
        "E309",
        "E501"
    ],
    "anaconda_linter_underlines": false,
    "anaconda_linter_mark_style": "none",
    "display_signatures": false,
    "disable_anaconda_completion": true,
    "python_interpreter": "/usr/local/bin/python3"
}
```

Also, autocomplete does not work. It is a clean installation of Sublime text 3. 

I found out through google that this line might help, but it did not:
`"auto_formatting_timeout": 5,`

So any ideas on how to fix this? Either autoformatting or autocomplete. Thanks.

**Comment by thymaro** [comment]
hi @neuro-01   has your issue been resolved in the meantime? if not, what other plugins and extensions have you got installed in ST3?

**Comment by neuro-01** [comment]
No, it did not. I just stopped using ST3 and start using PyCharm. It was pretty much clear installation. Only a few things, like BracketHighlighter, Material Theme, and Anaconda.

**Comment by meikel24** [comment]
i found the right answer to this. from the default settings find "auto_formatting": false, and set it to true then on user settings delete this "auto_formatting": true,.. and it should work fine

**Comment by AnkeEtz** [comment]
Are you sure the path to your python interpreter is correct. On Linux it's likely to be /usr/bin/python or /usr/bin/python3.

**Comment by RogueSoul7** [comment]
Sublime is looking for ```python``` in environmental variable when you build it, and it is not able to find it there. You can test it in cmd by typing ```python``` and if it opens Microsoft store then here is the fix you can do, it worked for me. Open setting -> search for ```manage app execution aliases``` -> turn off "python.exe" and "python3.exe". Now check if you can get python in cmd by typing ```python```, if you do then sublime problem should be fixed. If you want to know why this works follow this [stack overflow](https://stackoverflow.com/questions/58754860/cmd-opens-window-store-when-i-type-python)

**Comment by Blues1998** [comment]
> i found the right answer to this. from the default settings find "auto_formatting": false, and set it to true then on user settings delete this "auto_formatting": true,.. and it should work fine

this didn't work for me. still getting the same error.

**Comment by cs96665** [comment]
These are my configurations, I find it helpful to restart sublime after saving these in the user config for the anaconda package. 
{
    "auto_formatting": true,
    "pep8_ignore":
    [
        "E501"
    ],
    "anaconda_linter_underlines": true,
    "anaconda_linter_mark_style": "none",
    "display_signatures": false,
    "disable_anaconda_completion": true,
    "auto_formatting_timeout": 5,
    "python_interpreter": "C:/Users/<your user>/anaconda3/python.EXE"
}
Basically link your python interpreter; use command prompt and type where python (on windows) and which python (for mac) and put the path there. Also, I say it again restart your sublime.
